### PR TITLE
fix: #1494 Port AdminButton from gatsby to next.js

### PR DESCRIPTION
**Pull request by sonechca**
## Issue This PR Addresses
Fixed: #1494 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by tonyvugithub** [comment]
@sonechca : Hi there, we are in the process of reviewing these migrated to Next PRs. Just wonder if you are still interested to work on this as we would like you to make some changes to your PR. Please let us know in case you decided to drop this then we can assign the issue to others. Thank you.

File: src/frontend/next/src/contexts/User/UserContext/index.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { createContext } from 'react';
```
Comment (humphd) [inline src/frontend/next/src/contexts/User/UserContext/index.js:1]: Delete this file.

File: src/frontend/next/src/components/AdminButtons/AdminButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useContext } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { IconButton } from '@material-ui/core';
+import FlagIcon from '@material-ui/icons/Flag';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { UserStateContext } from '../../contexts/User/UserContext';
```
Comment (humphd) [inline src/frontend/next/src/components/AdminButtons/AdminButtons.tsx:6]: Once #1521 lands (hopefully soon), we can change this to:

```ts
import { useUser } from '../UserProvider';
```

File: src/frontend/next/src/components/AdminButtons/AdminButtons.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  iconSpan: {
+    paddingLeft: '10px',
+    paddingRight: '10px',
+  },
+}));
+
+function AdminButtons() {
+  const classes = useStyles();
+  const user = useContext(UserStateContext);
```
Comment (humphd) [inline src/frontend/next/src/components/AdminButtons/AdminButtons.tsx:20]: This can become:

```ts
const user = useUser();
```

**Review by humphd — CHANGES_REQUESTED** [review]
A couple of things to do, and also, please `git mv` the file src/frontend/next/src/components/AdminButtons/AdminButtons.tsx to src/frontend/next/src/components/AdminButtons.tsx (i.e., get rid of the directory).

**Comment by humphd** [comment]
Also, you need a rebase on `master` to pickup the `package.json` changes:

```
git checkout master
git pull upstream master
git checkout Port-AdminButtons
git rebase master
...fix conflicts...
git add src/frontend/next/package.json
git rebase --continue
```

**Comment by humphd** [comment]
@sonechca what's happening here?  I'd like to get this shipped today.

**Comment by humphd** [comment]
We decided to move this to 1.6, and we'll take the work in #1504 to add a dummy AdminButtons component for now.

**Comment by yuanLeeMidori** [comment]
close by #1619

**Pull request closed without merging**